padre_meddea/io/fits_tools.py

```diff
@@ -422,3 +435,745 @@ def get_prhsh(n: int = 1, a: str = "A") -> Tuple[str, str]:
+                            f"Using {len(new_table)-1} as end_idx."
+                        )
+                        end_idx = len(new_table) - 1
+
+                    if start_idx < 0:
+                        log.warning(
+                            f"File {source_path}: start_idx {start_idx} is negative. Using 0 as start_idx."
+                        )
+                        start_idx = 0
+
+                    # Select only the relevant rows
+                    selected_data = new_table[start_idx : end_idx + 1]
+
+                    # Stack with existing data
+                    hdu_dict[i]["data"] = vstack([hdu_dict[i]["data"], selected_data])
+
+                elif hdu_dict[i]["type"] == "image":
+                    # Concatenate image data (assuming along first axis), using only selected indices
+                    # Make sure indices are within valid range
+                    if end_idx >= hdu.data.shape[0]:
+                        log.warning(
+                            f"File {source_path}: end_idx {end_idx} exceeds image length {hdu.data.shape[0]}. "
+                            f"Using {hdu.data.shape[0]-1} as end_idx."
+                        )
+                        end_idx = hdu.data.shape[0] - 1
+
+                    if start_idx < 0:
+                        log.warning(
+                            f"File {source_path}: start_idx {start_idx} is negative. Using 0 as start_idx."
+                        )
+                        start_idx = 0
+
+                    # Select only the relevant slice of the image data
+                    selected_data = hdu.data[start_idx : end_idx + 1]
+
+                    # Concatenate with existing data
+                    hdu_dict[i]["data"] = np.concatenate(
+                        [hdu_dict[i]["data"], selected_data], axis=0
+                    )
+
+    return hdu_dict
+
+
+def _write_output_file(hdu_dict: dict, outfile: Path) -> None:
+    """
+    Construct and write the output FITS file.
+
+    Parameters
+    ----------
+    hdu_dict : dict
+        Dictionary containing HDU information
+    outfile : Path
+        Output file path
+    """
+    hdu_list = []
+    for i in sorted(hdu_dict.keys()):
+        hdu_info = hdu_dict[i]
+
+        if hdu_info["type"] == "primary":
+            hdu_list.append(fits.PrimaryHDU(header=hdu_info["header"]))
+        elif hdu_info["type"] == "bintable":
+            new_hdu = fits.BinTableHDU(
+                hdu_info["data"], header=hdu_info["header"], name=hdu_info["name"]
+            )
+            new_hdu.add_checksum()
+            hdu_list.append(new_hdu)
+        elif hdu_info["type"] == "image":
+            new_hdu = fits.ImageHDU(
+                hdu_info["data"], header=hdu_info["header"], name=hdu_info["name"]
+            )
+            hdu_list.append(new_hdu)
+
+    # Write the output file
+    hdul = fits.HDUList(hdu_list)
+    hdul.writeto(outfile, overwrite=True)
```

Comment:
We are not using a `with` in the function which could be causing the issue in Windows. Can we add a 
`hdul.close()` right after?